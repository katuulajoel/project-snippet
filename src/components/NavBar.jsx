/* -------------------------------------------------------------------------- */
/*                            External Dependencies                           */
/* -------------------------------------------------------------------------- */
import PropTypes from "prop-types";
import React from "react";
import { Link } from "react-router-dom";
import { withRouter } from "react-router-dom";
import styled from "styled-components";

const NavBar = ({ className, title }) => {
  return (
    <Wrapper className={`navbar fixed-top navbar-dashboard ${className || ""}`}>
      <Link to={`/dashboard`} className="navbar-brand">
        {title}
      </Link>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbar"
        aria-controls="navbar"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <i className="tg-ic-bars" />
      </button>
      <div className="collapse navbar-collapse" id="navbar"></div>
    </Wrapper>
  );
};

const Wrapper = styled.nav`
  padding-bottom: 10px;
  padding-top: 10px;
  background-color: #ffffff;
  box-shadow: 0 2px 4px 0 hsl(0deg 0% 80% / 50%);

  &.fixed-top {
    position: fixed;
    right: 0;
    left: 0;
    z-index: 1030;
  }

  &.navbar-dashboard {
    left: 220px;
    padding: 16px 40px;
    min-height: 73px;

    .navbar-brand {
      color: #151a30;
      font-weight: bold;
      line-height: 24px;

      &.hover {
        color: #000000;
      }
    }

    .navbar-nav {
      > li {
        > a {
          color: #000000;
        }
      }
    }
  }

  &.no-title-bar {
    box-shadow: 0 2px 4px 0 hsla(0, 0%, 80%, 0.5);
  }

  .navbar-brand {
    margin: 0;
    padding: 0;
    font-size: 20px;
    color: #ffffff;

    img {
      height: 50px;
    }
  }

  .navbar-nav {
    .input-search {
      @include padding-y(px-to-rem(7.5));
    }

    > li {
      > a {
        color: get-color("white");
        line-height: 50px;
        @include padding-x(px-to-rem(10));

        &:hover,
        &:focus {
          text-decoration: none;
        }

        .caret {
          margin-left: 5px;
        }
      }

      .dropdown-menu {
        min-width: 190px;
        left: auto;
        right: 0;
        margin-top: 0;
        @include padding-y(0);
        box-shadow: 0 2px 4px 0 rgba(204, 204, 204, 0.5);

        &.dropdown-menu-account {
          min-width: 100px;

          > li > a,
          > li > a:hover,
          > li > a:active {
            color: get-color("black");
            @include padding-y(px-to-rem(10));
            @include padding-x(px-to-rem(14));

            &:hover,
            &:focus {
              color: #da3451;
              background-color: get-color("white");
            }
          }
        }

        > li > a,
        > li > a:hover,
        > li > a:active {
          display: block;
          color: #da3451;
          @include padding-y(px-to-rem(5));
          @include padding-x(px-to-rem(10));

          &:hover,
          &:focus {
            text-decoration: none;
            color: get-color("white");
            background-color: #da3451;
          }
        }
      }
    }

    .btn.btn-auth {
      height: px-to-rem(40);
      line-height: px-to-rem(40);
      padding-left: 20px;
      padding-right: 20px;
      color: #da3451;
      background-color: get-color("light-grey");
      box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.5);
    }

    .btn-call {
      margin-right: 10px;
    }

    .avatar-wrapper {
      background-color: get-color("white");
      line-height: 1;

      .avatar {
        color: get-color("white");
        background-color: #da3451;
      }
    }
  }
`;

NavBar.propTypes = {
  title: PropTypes.string,
  className: PropTypes.string,
};

export default withRouter(NavBar);
