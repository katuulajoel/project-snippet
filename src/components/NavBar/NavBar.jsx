/* eslint-disable no-undef */
import PropTypes from "prop-types";
import React from "react";
import { Link } from "react-router-dom";
import styleable from "react-styleable";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import styled from "styled-components";

import { toggleRightNav } from "../../actions/UtilityActions";
// import { createResult } from "../../actions/TestResultsActions";

import Icon from "../core/Icon";
import SearchBox from "../core/SearchBox";
import Button from "../core/Button";
import { proxySafeUrl } from "../utils/proxy";

import { isAdminOrPM } from "../utils/auth";

// import { openModal } from "../core/utils/modals";

// import TestForm from "../../pages/Dashboard/tests/TestForm";

import {
  ButtonDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
} from "reactstrap";

import css from "./NavBar.module.scss";

class NavBar extends React.Component {
  static defaultProps = {
    variant: "dashboard",
    breakpoint: "md",
    isLargeDevice: false,
  };

  static propTypes = {
    title: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    variant: PropTypes.string,
    className: PropTypes.string,
    user: PropTypes.object,
    onSignOut: PropTypes.func,
    breakpoint: PropTypes.string,
    isLargeDevice: PropTypes.bool,
    noTitleBar: PropTypes.bool,
    location: PropTypes.object,
    history: PropTypes.object,
    collapseRightNav: PropTypes.func,
    Project: PropTypes.object,
    match: PropTypes.object,
    saveTestResult: PropTypes.func,
  };

  constructor(props) {
    super(props);
    this.state = {
      copied: false,
      dropdownOpen: false,
      isProjectDetail: false,
      projectId: null,
    };
  }

  componentDidMount() {
    this.hideCreateSearch(this.props.location);

    let updateNavbar = function () {
      let windowWidth = $(window).innerWidth();
      if (windowWidth >= 768) {
        if ($(document).scrollTop() >= 20) {
          $(".navbar").addClass("navbar-showcase-fixed");
        } else {
          $(".navbar").removeClass("navbar-showcase-fixed");
        }
      }
    };

    const { variant } = this.props;
    if (variant === "showcase") {
      $(document).ready(updateNavbar);
      $(document).scroll(updateNavbar);
      $(window).resize(updateNavbar);
    }

    this.unlisten = this.props.history.listen((location) => {
      this.hideCreateSearch(location);
    });
  }

  componentWillUnmount() {
    this.unlisten();
  }

  onSignOut(e) {
    if (e) {
      e.preventDefault();
    }
    if (this.props.onSignOut) {
      this.props.onSignOut();
    }
  }

  setCopied = () => {
    this.setState({ copied: true });
  };

  clearCopied = () => {
    let self = this;
    // user timer to prevent flicker
    setTimeout(() => {
      if (self.state.copied) {
        this.setState({ copied: false });
      }
    }, 1000);
  };

  toggle = () => {
    this.setState({ dropdownOpen: !this.state.dropdownOpen });
  };

  hideCreateSearch = (location) => {
    let projectId = location.pathname.split("/")[2];
    if (!isNaN(projectId)) {
      this.setState({ isProjectDetail: true, projectId: projectId });
    } else {
      this.setState({ isProjectDetail: false });
    }
  };

  openRightNav = () => {
    this.props.collapseRightNav(true, "project");
  };

  //   addNewTest = () => {
  //     openModal(
  //       <TestForm id="test-form" />,
  //       `Add New Result`,
  //       true,
  //       {
  //         className: "modal-tests",
  //         ok: `Save`,
  //         cancel: "Close",
  //         form: {
  //           type: "submit",
  //           form: `test-form`,
  //         },
  //         style: { maxWidth: "768px" },
  //       },
  //       null,
  //       false
  //     ).then(
  //       (data) => {
  //         this.props.saveTestResult(data);
  //       },
  //       () => {}
  //     );
  //   };

  render() {
    let {
      title,
      user,
      variant,
      breakpoint,
      className,
      noTitleBar,
      //   Project: { projects },
    } = this.props;

    return (
      <nav
        className={`navbar navbar-expand-${
          breakpoint || "md"
        } fixed-top navbar-dark ${className || ""} ${
          variant ? `navbar-${variant}` : ""
        } ${noTitleBar ? "no-title-bar" : ""}`}
      >
        <Link
          to={`/${variant === "dashboard" ? "dashboard" : ""}`}
          className="navbar-brand"
        >
          {variant === "dashboard" ? (
            title
          ) : (
            <img src={require("../../assets/images/logo_new_color.png")} />
          )}
        </Link>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbar"
          aria-controls="navbar"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <i className="tg-ic-bars" />
        </button>
        <div className="collapse navbar-collapse" id="navbar">
          <ul className="navbar-nav ml-auto">
            {user && user.id ? (
              this.state.isProjectDetail ? (
                projects[this.state.projectId] &&
                projects[this.state.projectId].stage !== "opportunity" ? (
                  <li>
                    <Button
                      className={css["right-nav-button"]}
                      onClick={() => {
                        this.openRightNav();
                      }}
                    >
                      <Icon name="dots-horizontal" />
                      &nbsp;&nbsp;&nbsp;More
                    </Button>
                  </li>
                ) : null
              ) : (
                <>
                  {variant !== "showcase" && this.props.match.url !== "/tests" && (
                    <li>
                      <StyledSearchBox
                        variant="search"
                        clear={true}
                        match={this.props.match}
                      />
                    </li>
                  )}
                  {!isAdminOrPM() ? null : this.props.match.url === "/tests" ? (
                    <li>
                      <StyledButton onClick={() => this.addNewTest()}>
                        <Icon name="round-add" />
                        &nbsp;&nbsp;&nbsp;Add New Result
                      </StyledButton>
                    </li>
                  ) : (
                    <li className={css["add-item"]}>
                      <ButtonDropdown
                        className={css["add-btn"]}
                        isOpen={this.state.dropdownOpen}
                        toggle={this.toggle}
                      >
                        <DropdownToggle>
                          <Icon name="round-add" />
                          &nbsp;&nbsp;&nbsp;New
                        </DropdownToggle>
                        <DropdownMenu className={css["dropdown-menu"]}>
                          <DropdownItem className={css["dropdown-item"]}>
                            <Link to="/projects/new/managed/details">
                              <Icon name="briefcase-clock-outline" />
                              Managed Project
                            </Link>
                          </DropdownItem>
                          <DropdownItem className={css["dropdown-item"]}>
                            <Link to="/projects/new/dedicated/details">
                              <Icon name="account-outline" />
                              Dedicated Developer
                            </Link>
                          </DropdownItem>
                          <DropdownItem className={css["dropdown-item"]}>
                            <Link to="/projects/new/opportunity">
                              <Icon name="bullhorn-outline" />
                              Opportunity
                            </Link>
                          </DropdownItem>
                        </DropdownMenu>
                      </ButtonDropdown>
                    </li>
                  )}
                </>
              )
            ) : (
              <>
                <li>
                  <Link to={proxySafeUrl("/signin")} className="btn btn-auth">
                    Login
                  </Link>
                </li>
              </>
            )}
          </ul>
        </div>
      </nav>
    );
  }
}

const StyledButton = styled(Button)`
  box-shadow: none;
`;

const StyledSearchBox = styled(SearchBox)`
  border: 1px solid #edf1f7;
  box-sizing: border-box;
  border-radius: 4px;
  padding: 7.5px 18px;

  .input-group-prepend {
    padding-right: 13px;

    .input-group-text {
      background-color: #fff;
      color: #3e4857;
      padding: 0;
      font-size: 16px;
      height: initial;
    }
  }

  .input-group-prepend:last-child {
    padding-left: 8px;
    padding-right: 0px;
    button {
      background-color: #fff;
      padding: 0;
      height: initial;
      i {
        color: #8f9bb3;
        font-size: 16px;
      }
    }
  }

  > input {
    background-color: #fff;
    color: #3e4857;
    padding: 0 !important;
    font-size: 16px;
    height: auto !important;
    &:focus {
      background-color: #fff;
    }
  }
`;

function mapStateToProps(state) {
  return {
    Project: state?.app?.Project,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    collapseRightNav: bindActionCreators(toggleRightNav, dispatch),
    // saveTestResult: bindActionCreators(createResult, dispatch),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(styleable(css)(NavBar)));
