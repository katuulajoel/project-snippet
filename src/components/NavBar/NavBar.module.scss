// Variables: includes Bootstrap overrides
@import "../../scss/variables.scss";

.add-item {
    margin-left: 24px;

    .add-btn {
        > button {
            font-weight: 600;
            font-size: px-to-rem(16);
            line-height: px-to-rem(22);
            text-align: center;
            background: #da3451;
            border-color: transparent;
            box-shadow: none;
            border-bottom-right-radius: 4px !important;
            border-top-right-radius: 4px !important;
        }
    }
}

.right-nav-button {
    color: get-color("text-grey");
    background: rgba(143, 155, 179, 0.1);
    border: none;
    border-radius: 6px;
    box-shadow: none;
    line-height: px-to-rem(22);
    font-size: px-to-rem(14);
    text-align: right;

    i {
        font-size: 3px;
    }

    &:hover,
    &:visited,
    &:link,
    &:not(:disabled):not(.disabled):active,
    &:not(:disabled):not(.disabled):active:focus,
    &:focus {
        outline: 0;
        box-shadow: none;
        border: none;
        border-radius: 6px;
        color: get-color("text-grey");
        background: rgba(143, 155, 179, 0.2);
    }
}

/*override reactstrap inline styles*/
.dropdown-menu {
    transform: translate3d(0px, 54px, 0px) !important;
    left: initial !important;
    border: 1px solid #edf1f7 !important;
    box-shadow: 0px 4px 8px rgba(62, 72, 87, 0.04) !important;
    border-radius: 6px !important;
    padding: 10px !important;

    .dropdown-item {
        padding: 0;
        color: #3e4857;

        &:active {
            color: initial;
            background-color: initial;
        }

        a{
            padding: 15px;
            color: #3e4857;
            text-decoration: none;
            display: block;
            i{
                margin-right: 12px;
            }
        }
    }
}
