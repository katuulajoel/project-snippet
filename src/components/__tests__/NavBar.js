import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import NavBar from "../NavBar";

describe("Auth layout test", () => {
  it("Snapshot test for Login component", () => {
    const tree = renderer
      .create(
        <Router>
          <NavBar />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
