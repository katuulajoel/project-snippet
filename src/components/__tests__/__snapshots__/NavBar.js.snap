// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth layout test Snapshot test for Login component 1`] = `
<nav
  className="sc-bdnxRM cZRQnq navbar fixed-top navbar-dashboard "
>
  <a
    className="navbar-brand"
    href="/dashboard"
    onClick={[Function]}
  />
  <button
    aria-controls="navbar"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbar"
    data-toggle="collapse"
    type="button"
  >
    <i
      className="tg-ic-bars"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbar"
  />
</nav>
`;
